src/config/backends/extlinux-file.ts

```diff
@@ -0,0 +1,86 @@
+import * as _ from 'lodash';
+
+import { ConfigOptions } from './backend';
+import { ExtLinuxParseError } from '../../lib/errors';
+
+export interface ExtlinuxFile {
+	labels: Label;
+	globals: Directive;
+}
+
+export interface Label {
+	[labelName: string]: Directive;
+}
+
+export interface Directive {
+	[directive: string]: string;
+}
+
+export abstract class ConfigurableDirective {
+	// Parses the values for this directive
+	public abstract parse(directives: Directive): ConfigOptions;
+	// Return the value to be set for this directive using the provided ConfigOptions
+	public abstract value(opts: ConfigOptions, existingValue: string): string;
+}
+
+export class AppendDirective extends ConfigurableDirective {
+	public parse(directives: Directive): ConfigOptions {
+		const appendEntry = directives.APPEND;
+
+		if (appendEntry == null) {
+			throw new ExtLinuxParseError(
+				'Could not find APPEND directive in default extlinux.conf boot entry',
+			);
+		}
+
+		const configOpts: ConfigOptions = {};
+		const values = appendEntry.split(' ');
+
+		for (const value of values) {
+			const parts = value.split('=');
+			if (parts.length !== 2) {
+				configOpts[parts[0]] = '';
+			} else {
+				configOpts[parts[0]] = parts[1];
+			}
+		}
+
+		return configOpts;
+	}
+
+	public value(opts: ConfigOptions, existingValue: string): string {
+		// Parse current append line and remove already set values so we don't have duplicates
+		const appendLine = _.filter(existingValue.split(' '), (entry) => {
+			const lhs = entry.split('=');
+			return !_.includes(opts, lhs[0]);
+		});
+
+		// Apply new configurations values to the provided append line
+		_.each(opts, (value, key) => {
+			if (!value) {
+				// Example: rootwait (config without a value)
+				appendLine.push(`${key}`);
+			} else {
+				// Example: isolcpus=2,3 (config with a value)
+				appendLine.push(`${key}=${value}`);
+			}
+		});
+
+		// Return modified line as a string
+		return appendLine.join(' ');
```

Comment:
I prefer a map/concat style as it makes it clear that we're mapping over/transforming the array in a direct 1-1 style
```suggestion
		return appendLine.concat(_.map(opts, (value, key) => {
			if (!value) {
				// Example: rootwait (config without a value)
				return `${key}`;
			} else {
				// Example: isolcpus=2,3 (config with a value)
				return `${key}=${value}`;
			}
		})).join(' ');
```

Reply:
I agree and added this implementation.